mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -172,14 +172,11 @@ class StringIndexerModel private[ml] (
   }
 
   /**
-   * Return a model to perform the inverse transformation.
-   * Note: By default we keep the original columns during this transformation, so the inverse
-   * should only be used on new columns such as predicted labels.
+   * Return a [[IndexToString]] instance to perform the inverse transformation, mapping indices back
+   * to their string values.
```

Comment:
I think it's useful to provide some comment to the effect that this inverse should be used to convert predicted indices back to labels (though I agree the wording can be improved).